rust/lanzatool/src/esp.rs

```diff
@@ -3,6 +3,41 @@ use std::path::{Path, PathBuf};
 
 use crate::generation::Generation;
 
+#[allow(dead_code)]
+#[non_exhaustive]
+pub enum EfiFallback {
+    X86,
+    Arm,
+    AArch64
+}
+
+// TODO: rethink the API design
```

Comment:
Yes. :laughing: 

Maybe we can just piggy back on [ProcessorArch](https://docs.rs/uefi/0.18.0/uefi/proto/debug/struct.ProcessorArch.html). This would avoid the awkwardly named `EfiFallback`.

Then we can just have some freestanding functions:

- `system_double_to_efi_arch(&str) -> ProcessorArch`
- `efi_fallback_filename(ProcessorArch) -> &Path`
- `systemd_filename(ProcessorArch) -> &Path`

PS. Technically a 32-bit UEFI system can host a 64-bit operating system. This happened on old Atom tablets... So technically a `x86_64-linux` system could be either 32 or 64-bit UEFI. In reality though, 32-bit UEFI should be pretty dead on x86.

Reply:
I have noticed the 32-bits UEFI system thing, but I am… tempted to not support it at all, except if someone come with a PR and a test.

Agreed on the design.

Reply:
Maybe we should just stick with the nix terminology and call it `System`? Although I personally hate the term "System", but maybe we should do it for consistency's sake. 

This could also be implemented upstream in bootspec, so the system field is more strongly typed. We could then just implement a trait for `System` like `impl SystemdBootPaths for bootspec::System`

Reply:
`System` is good. I don't see why we would need traits, but ymmv.

Reply:
I guess it should be `systemd_filename(&self) -> Option<&Path>` because I'm not sure sd-boot supports all platforms, same goes for EFI fallback filenames. (does SPARC64 have UEFI support?)

Reply:
> `System` is good. I don't see why we would need traits, but ymmv.

We would only need a trait if we want to implement something on a foreign type (i.e. a `System` enum from the bootspec lib).

Reply:
I implemented what we said, do not hesitate to reopen if not satisfying.